static/src/javascripts/projects/commercial/modules/dfp/prepare-permutive.spec.js

```diff
@@ -64,80 +64,104 @@ describe('Generating Permutive payload utils', () => {
     describe('generatePayload', () => {
         it('safely removes invalid authors or keywords', () => {
             const invalidValues = [undefined, null, '', {}, []];
-            const expected = {};
+            const expected = { user: { identity: false } };
```

Comment:
This is a great example of why in tests it's nice to test a specific property rather than check the whole return value. If your test concerns the keyword and author properties, better to only check those, so these tests don't need refactoring every time anything else changes.

When taking the above approach, it's often useful to have another test that explicitly checks the whole object is as expected to maintain a "big picture" test of the whole data structure. It's no huge hassle to update this in one when changes are made, rather than having to update the tests for every property separately.

Not a huge deal here and no pressure to change this, but just wanted to share the approach I've adopted in these kinds of cases :-)

Reply:
Yep makes total sense 👍